### PR TITLE
Systemd cookbook

### PR DESCRIPTION
**Pull request by olive42**

### CODE REVIEW

File: libraries/nexus_helper.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+The Nexus server at #{endpoint} did not become ready within #{timeout}
+seconds. On large Nexus instances (or on smaller hardware), you may need
+to increase the timeout to #{timeout * 4} seconds. Alternatively, Nexus
+may have failed to start.
+EOH
+      end
+    end
+
+    def wait_until_ready!(endpoint, timeout = 15 * 60)
+      Timeout.timeout(timeout, Timeout::Error) do
```
Comment (jmauro) [inline libraries/nexus_helper.rb]: OPT: `::Timeout`

File: resources/default.rb [hunk 2]
```diff
@@ -1,18 +1,125 @@
...
+property :path, kind_of: String, default: lazy { node['nexus3']['path'] }
+property :data, kind_of: String, default: lazy { node['nexus3']['data'] }
+# Name of service used as property for nexus3_service; can be instance_name or something different
+# like version number.
+property :service_name, kind_of: String, default: lazy { instance_name }
+property :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
+property :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+
+action :install do
+  install_dir = ::File.join(new_resource.path, "nexus-#{version}")
```
Comment (jmauro) [inline resources/default.rb]: - This variable is the same as a property could change the name plz?
- `new_resource.version`

File: resources/service_sysvinit.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+action_class do
+  def create_init
+    link ::File.join('/etc/init.d', "nexus3_#{new_resource.instance_name}") do
+      to ::File.join(new_resource.install_dir.to_s, 'bin', 'nexus')
+      owner new_resource.nexus3_user
+      group new_resource.nexus3_group
+    end
+  end
+
+  def platform_sysv_init_class
```
Comment (jmauro) [inline resources/service_sysvinit.rb]: Function unused

File: libraries/nexus_helper.rb (hunk 1 above)
Comment (olive42) [reply libraries/nexus_helper.rb]: done

File: resources/default.rb (hunk 2 above)
Comment (olive42) [reply resources/default.rb]: done

**Pull request closed without merging**